### PR TITLE
Do not change homepage tab at opening if an item from the asset store…

### PR DESCRIPTION
**Pull request by AlexandreSi**
… is requested

### CODE REVIEW

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js [hunk 1]
```diff
@@ -173,26 +178,57 @@ export const HomePage = React.memo<Props>(
...
+          if (isShopRequestedAtOpening.current) return;
           if (shouldChangeTabAfterUserLoggedIn.current) {
             setActiveTab(authenticated ? initialTab : 'get-started');
           }
         },
         // Only the initialTab at component mounting is interesting
         // and we don't want to change the active tab if initialTab changes.
         // eslint-disable-next-line react-hooks/exhaustive-deps
-        [authenticated]
+        [authenticated, routeArguments]
```
Comment (ClementPasteau) [inline newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js:231]: is this needed?

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js [hunk 2]
```diff
@@ -173,26 +178,57 @@ export const HomePage = React.memo<Props>(
...
+        [
+          routeArguments,
+          removeRouteArguments,
+          setInitialPackUserFriendlySlug,
+          setInitialGameTemplateUserFriendlySlug,
+        ]
       );
 
       // If the user is not authenticated, the GetStarted section is displayed.
       React.useEffect(
```
Comment (ClementPasteau) [inline newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js:221]: This effect scares me a bit as it really feels like tinkering.
It'll probably bit us back when we'll add new initial-tab options, but I guess we're not there yet

**Review by ClementPasteau — APPROVED** [review]

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js (hunk 2 above)
Comment (AlexandreSi) [reply newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js:221]: I understand but it's a symptom of the design, I think that's the cleanest way to accomplish the objective of the effect

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js:231]: Good catch!

**Pull request merged**